tensorbay/opendataset/UAVDT/loader.py

```diff
@@ -0,0 +1,151 @@
+                    M0101_gt_whole.txt
+                    ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = Dataset(_DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for mode in _MODES:
+        for sequence_attributes_path in glob(os.path.join(root_path, "M_attr", mode, "*.txt")):
+            sequence = os.path.basename(sequence_attributes_path).split("_", 1)[0].strip()
+            segment = dataset.create_segment(f"{mode}-{sequence}")
+
+            sequence_attributes = _convert_attr_txt_to_dict(
+                sequence_attributes_path, dataset.catalog.classification.attributes.keys()
+            )
+            classification = Classification(attributes=sequence_attributes)
+
+            frame_id_ground_truth_map = _convert_gt_txt_to_dict(
+                root_path,
+                sequence,
+                dataset.catalog.box2d,
+            )
+
+            image_paths = glob(os.path.join(root_path, "UAV-benchmark-M", sequence, "*.jpg"))
+            for image_path in image_paths:
+                data = Data(image_path)
+                ground_truths = frame_id_ground_truth_map[_extract_frame_id(image_path)]
+                data.label.classification = classification
+                data.label.box2d = _convert_ground_truths_to_labels(ground_truths)
+                segment.append(data)
+
+    return dataset
+
+
+def _convert_attr_txt_to_dict(
+    sequence_attributes_path: str, attribute_names: Tuple[str, ...]
+) -> Dict[str, bool]:
+    with open(sequence_attributes_path) as fp:
+        csv_reader = csv.reader(fp)
+        elements = next(csv_reader)
+        attributes = {
+            attribute_name: bool(int(value))
+            for attribute_name, value in zip(attribute_names, elements)
+        }
+
+    return attributes
+
+
+def _convert_gt_txt_to_dict(
+    path: str,
+    sequence: str,
+    box2d_catalog: Box2DSubcatalog,
+) -> Dict[int, List[Dict[str, Any]]]:
+    attributes = box2d_catalog.attributes
+    catagories = box2d_catalog.categories.keys()
+
+    ground_truth_path = os.path.join(path, "UAV-benchmark-MOTD_v1.0", "GT")
+    frame_id_ground_truth_map = defaultdict(list)
+    with open(os.path.join(ground_truth_path, f"{sequence}_gt.txt")) as file_mot, open(
+        os.path.join(ground_truth_path, f"{sequence}_gt_whole.txt")
+    ) as file_det:
+        mot_csv = csv.reader(file_mot)
+        det_csv = csv.reader(file_det)
+        for elements_mot, elements_det in zip(mot_csv, det_csv):
+            ground_truth: Dict[str, Any] = {}
+            ground_truth["target_id"] = elements_det[1]
+            ground_truth["bbox_xywh"] = map(float, elements_det[2:6])
+            ground_truth["bbox_attr"] = {
```

Comment:
in our SDK, `attr` means python object attribute, `attribute` means label attribute.